hermit-sys: Network regression

I noticed this issue while rebasing #124.

While all updates on master since opening the PR bc3db32...fc6de5c seem to be unrelated to virtio, I can't get the netbench to work anymore.

## fc6de5c

In both debug and release I see two ICMPv6 Multicast Listener Report Messages v2 (143) with Wireshark, but no ARP request for `10.0.5.1`.
Strange that there are different outputs on debug and release...

### debug
```
[0][INFO] Feature set wanted by network driver are in conformance with specification.
[0][ERROR] Device features set, does not satisfy minimal features needed. Aborting!
[0][ERROR] Virtio networkd driver could not be initialized with device: 1041
[INFO] Spawn network thread with id 2
[WARN] Ethernet interface not available
Connecting to the server 10.0.5.1...
Couldn't connect to server...
```

### release
```
[0][INFO] Feature set wanted by network driver are in conformance with specification.
[0][INFO] Feature set wanted by network driver are in conformance with specification.
[0][INFO] Driver found a subset of features for virtio device 1041. Features are: [VIRTIO_NET_F_MAC, VIRTIO_NET_F_STATUS, VIRTIO_F_RING_INDIRECT_DESC, VIRTIO_F_VERSION_1, VIRTIO_F_RING_PACKED]
[0][INFO] Features have been negotiated between virtio network device 1041 and driver.
[0][INFO] Created PackedVq: idx=0, size=256
[0][INFO] Created PackedVq: idx=1, size=256
[0][INFO] Network driver successfully initialized virtqueues.
[0][INFO] Device specific initialization for Virtio network defice 1041 finished
[0][INFO] Network device with id 1041, has been initialized by driver!
[0][INFO] Virtio-net link is up after initialization.
[0][INFO] Virtio network driver initialized with Virtio network device.
[0][INFO] Install virtio interrupt handler at line 11
[INFO] Spawn network thread with id 2
[INFO] MAC address 52-54-00-12-34-56
[INFO] Configure network interface with address 10.0.5.3/24
[INFO] Configure gateway with address 10.0.5.1
[INFO] MTU: 1500 bytes
Connecting to the server 10.0.5.1...
Couldn't connect to server...
```

## bc3db32 (debug/release)
```
[0][INFO] Feature set wanted by network driver are in conformance with specification.
[0][INFO] Feature set wanted by network driver are in conformance with specification.
[0][INFO] Driver found a subset of features for virtio device 1041. Features are: [VIRTIO_NET_F_MAC, VIRTIO_NET_F_STATUS, VIRTIO_F_RING_INDIRECT_DESC, VIRTIO_F_VERSION_1]
[0][INFO] Features have been negotiated between virtio network device 1041 and driver.
[0][INFO] Created SplitVq: idx=0, size=256
[0][INFO] Created SplitVq: idx=1, size=256
[0][INFO] Network driver successfully initialized virtqueues.
[0][INFO] Device specific initialization for Virtio network defice 1041 finished
[0][INFO] Network device with id 1041, has been initialized by driver!
[0][INFO] Virtio-net link is up after initialization.
[0][INFO] Virtio network driver initialized with Virtio network device.
[0][INFO] Install virtio interrupt handler at line 11
[INFO] Spawn network thread with id 2
[INFO] MAC address 52-54-00-12-34-56
[INFO] Configure network interface with address 10.0.5.3/24
[INFO] Configure gateway with address 10.0.5.1
[INFO] MTU: 1500 bytes
Connecting to the server 10.0.5.1...
Connection established! Ready to send...
Sent everything!
```

--- a/hermit-sys/build.rs
+++ b/hermit-sys/build.rs
@@ -90,6 +90,7 @@ fn build_hermit(src_dir: &Path, target_dir_opt: Option<&Path>) {
 	}
 
 	let mut rustflags = Vec::new();
+	rustflags.push("-Zmutable-noalias=no".to_string());
 
 	#[cfg(feature = "instrument")]
 	{

--- a/hermit-sys/build.rs
+++ b/hermit-sys/build.rs
@@ -89,17 +89,31 @@ fn build_hermit(src_dir: &Path, target_dir_opt: Option<&Path>) {
 		cmd.arg("vga");
 	}
 
+	let mut rustflags = Vec::new();
+
 	#[cfg(feature = "instrument")]
 	{
-		cmd.env("RUSTFLAGS", "-Z instrument-mcount");
-		// if instrument is not set, ensure that instrument is not in environment variables!
-		cmd.env(
-			"RUSTFLAGS",
-			env::var("RUSTFLAGS")
-				.unwrap_or_else(|_| "".into())
-				.replace("-Z instrument-mcount", ""),
-		);
+		rustflags.push("-Zinstrument-mcount".to_string());
+		// Add outer `RUSTFLAGS` to command
+		if let Ok(var) = env::var("RUSTFLAGS") {
+			rustflags.push(var);
+		}
 	}
+
+	#[cfg(not(feature = "instrument"))]
+	{
+		// If the `instrument` feature feature is not enabled,
+		// filter it from outer `RUSTFLAGS` before adding them to the command.
+		if let Ok(var) = env::var("RUSTFLAGS") {
+			let flags = var
+				.split(',')
+				.filter(|&flag| !flag.contains("instrument-mcount"))
+				.map(String::from);
+			rustflags.extend(flags);
+		}
+	}
+
+	cmd.env("RUSTFLAGS", rustflags.join(","));
 
 	let output = cmd.output().expect("Unable to build kernel");
 	let stdout = std::string::String::from_utf8(output.stdout);

